### ISSUE TITLE
on_project_start: tmux -CC attach doesn't working

### ISSUE THREAD

**Issue by anquegi** [tmuxinator/tmuxinator#581]
Hi I have the following yml

```
# ~/.tmuxinator/pre-afs.yml

name: pre-afs
root: ~/

on_project_start: tmux -CC attach pre-afs

windows:
  - server_afs:
      layout: main-horizontal
      panes:
        - ssh pre-afs-fe-01
        - ssh pre-afs-fe-02
```
it works well and I get this:

<img width="1280" alt="captura de pantalla 2017-12-03 a las 10 52 58" src="https://user-images.githubusercontent.com/2518346/33524178-255b6eb4-d818-11e7-8295-8aebddd9b5d2.png">

but to get use of Iterm2 facilitities, I need to use:

C-b d
and write in tew terminal

tmux -CC attach -t pre-afs

then I finally get:

<img width="1280" alt="captura de pantalla 2017-12-03 a las 10 54 46" src="https://user-images.githubusercontent.com/2518346/33524206-6453a686-d818-11e7-97c4-ddab026477ac.png">

Is not supposed, that 

`on_project_start: tmux -CC attach pre-afs #or without pre-afs or with -t`

it is going to make this working

**Comment by ShirajG** [comment]
Same issue, I used to be able to jump into the native iterm2 integrated tmux mode using tmux -CC attach -t sessionName but now it will not work.

**Comment by ebalsumgo** [comment]
yeah i am having the exact same problem... if i switch to using post: tmux -CC attach it works as expected but tmuxinator throws an error because post: has been deprecated...

**Comment by gnapse** [comment]
Same here. Any news on this one? The deprecated option works, but the warning is annoying. The supposed replacement for it is not working with the `tmux -CC attach` command.

**Comment by ethagnawl** [comment]
I'll need to dig into this some more, but at first glance, it'd seem that if `post` works in this scenario then `project_exit` should also work.

This project config file results in the debug output posted below:

```
project_name: x
on_project_exit: echo "project exit"
post: echo "post"
windows:
  - shell:
    panes:
      - echo hello
```

```
λ tmuxinator debug x
#!/bin/bash

# Clear rbenv variables before starting tmux
unset RBENV_VERSION
unset RBENV_DIR

tmux start-server;

cd .
  
  # Create the session and the first window. Manually switch to root
  # directory if required to support tmux < 1.9
  TMUX= tmux new-session -d -s x -n shell
  tmux select-window -t x:0
  tmux select-pane -t x:0.0

  if [ -z "$TMUX" ]; then
    tmux -u attach-session -t x
  else
    tmux -u switch-client -t x
  fi

echo "post"

# Run on_project_exit command.
echo "project exit"
```

The "post" and "project exit" commands are executed one after the other.

The naming of `on_project_exit` might be confusing in this use case, but I'm not entirely clear on why it's working (as you all claim it is) in the first place -- those commands should only be executed after you've detached from/killed the session.

**Comment by endorama** [comment]
Hello all, this has been reported also in another issue, and the problem lies in the documentation which suggest using `on_project_start` instead of the correct `on_project_exit` with `attach: false`.

I just made a PR to fix this https://github.com/tmuxinator/tmuxinator/pull/650

From the configuration posted by @anquegi I would suggest setting `attach: false` in order to override the attaching command run by tmuxinator, and specifying this command in the `on_project_exit` hook.

`on_project_exit` is confusing in this use case, but is supposed to run after detaching from the tmux session ( which is similar to the confusion produced by the `post` configuration when `attach` is false ). @ethagnawl I would open a new issue to add some sort of `custom_attach_command` in order to help using this feature, what do you think?

Meanwhile could you try using the suggested workaround and report here? Thank you.

**Comment by ebalsumgo** [comment]
That seems to have done the trick!
